Mutect2 GATK4 with filtering and Panel-Of-Normals

szilvajuhos opened the pull request:
## PR checklist
 - [x] PR is made against `dev` branch
 - [x] This comment contains a description of changes
 - [ ] Ensure the test suite passes (`./scripts/test.sh -p docker -t ALL`).
 - [x] Documentation in `docs` is updated
 - [x] `CHANGELOG.md` is updated

Commit:
PON added to primary calls

Commit:
Path problems

Commit:
Merge remote-tracking branch 'upstream/dev'

Commit:
removed uneeded dbSNP reference map and added container for FilterMutect2Calls

Commit:
added container path also

Commit:
comment removed

Commit:
commit before refactoring to use threads

Commit:
works with sample data

Commit:
Mutect2 call and filtering separated

Commit:
params still not working

Commit:
added common SNPs to igenomes

Commit:
common SNPs and work with and without PON

Commit:
backout for scatter-gather

Commit:
removing useless --native-pair-hmm-threads, go for scatter-gather with MergeMutectStats

Commit:
searching from bug by diff

maxulysse committed:
Initial template commit

maxulysse committed:
WIP towards nf-core

Commit:
MuTect2 -> Mutect2

Commit:
config changes, profile for a general 16 CPUs node, also now we can merge Mutect2 stats

Commit:
Pileup summaries works

Commit:
CalculateContamination works

maxulysse committed:
update README (#1)

maxulysse committed:
More fixes (#2)

- update README
- update `.travis.yml`
- add `Jenkinsfile`
- update conda `environment.yml` file

maxulysse committed:
Include preprocessing (#3)

* update README
* update .travis.yml
* add Jenkinsfile
* update conda environment file
* update software_version collect
* remove lib/SarekUtils.groovy
* add References to .gitignore
* improve software versions gathering
* fix docker owner
* update tests
* sort params
* add preprocessing
* improve multiple TSV
* add parrallelized BaseRecalibrator

Commit:
Filtering works, but scatter-gather pileup generation is needed

maxulysse committed:
Add some Variant Calling (#4)

* Update allelecount to 4.0.2
* Update GATK to 4.1.2.0
* Update docs
* Add HaplotypeCaller
* Add single Strelka and single Manta

maxulysse committed:
fix Jenkins tests

maxulysse committed:
sort ansi codes

maxulysse committed:
add tests

maxulysse committed:
add Manta, Strelka, StrelkaBP, MuTecT2, Freebayes, Ascat, Controlfreec

maxulysse committed:
update docs

maxulysse committed:
fix travis tests

apeltzer committed:
Merge pull request #5 from MaxUlysse/nf-core-VC

Add somatic Variant Callers

maxulysse committed:
Add default containers for annotation (#6)

* add containers for annotation

maxulysse committed:
Nf core vc (#7)

* Add Annotation
* Annotation containers are tagged version.genomeversion
* Better indentation
* Replace `--sampleDir` by `--sample`
* add MultiQC
* update Tests
* Add parallelized testing on Travis
* Add parallelized building of Docker Images for annotation on CircleCI
* Update docs

Commit:
Pileup and filtering works with scatter-gather now

Commit:
Merge remote-tracking branch 'upstream/dev'

Commit:
Config for a general 16 CPUs, 96G mem machine

Commit:
initial commit

Commit:
tab->space

maxulysse committed:
Improve tests and docs (#8)

* improve tests + add some docs

Commit:
Mutect2 - related changes

Commit:
Merge remote-tracking branch 'upstream/dev'

Commit:
no TMP in recalibration, and fixes to run without PON

Commit:
Sarek-data updated with common SNPs for Mutect2 filtering

Commit:
tests are now not decompressing common SNPs

Commit:
adding accidentaly deleted line

szilvajuhos commented:
The test does not pass, stops at snpEff annotation. @MaxUlysse looks like a docker/container issue. Is it me only, or do you have that one also?

MaxUlysse commented:
There's definitively an issue on scilifelab/sarek, but not on nf-core/sarek.
Don't really understand why, I'll try to have a quick look at it, but as your stuff is working, and we're moving to nf-core/sarek, I prefer not to spend to much time as it doesn't seem critical yet

Commit:
again, fixing accidentaly deleted stuff

Commit:
Added default GnomAD allele frequencies

maxulysse committed:
more docs (#9)

* update tests
* code polishing
* add some docs about QC tools

maxulysse committed:
Docs (#11)

* better docs
* add docs about annotation
* move helper scripts to scripts
* add helper script to filter FreeBayes output
* add docs about Variant Calling and Preprocessing
* add scripts for ASCAT
* fix publishDir for MergeMpileup
* fix intra link
* completed docs about ASCAT
* add building container for dogs
* add Abstracts & Posters
* improve tests
* update docs

maxulysse committed:
Configs and docs (#12)

* Update CI scripts
* Add possibility to download singularity images with helper script
* Update Docs
* Improve build.nf
* Use label for processes configuration
* Update configuration files
* Disable Docker in singularity profile
* Disable Singularity in docker profile
* Disable Docker and Singularity in conda profile
* Simplify check_max() function

maxulysse committed:
Fix merged annotation (#13)

* Add BamQC, CompressVCFsnpEff and CompressVCFvep processes
* Add Citation documentation
* Fix merge in annotation
* Merge BamQCmapped and BamQCrecalibrated processes into BamQC process
* Removed BamQCmapped, BamQCrecalibrated and CompressVCF processes
* Split CompressVCF process into CompressVCFsnpEff and CompressVCFvep processes

maxulysse committed:
fix output name for vcf

apeltzer committed:
Merge pull request #14 from MaxUlysse/nf-core-dev

fix output name for vcf

maxulysse committed:
Fix R scripts (#16)

* chmod +x
* add correct path to Rscript
* update CHANGELOG

Commit:
Merge remote-tracking branch 'upstream/dev'

maxulysse committed:
More improvments (#18)

* Improve cpu usage
* Use same font for nf-core and sarek in ascii art
* Improve syntax
* Add --no-reports option for tests + add snpEff,VEP,merge to MULTIPLE test
* Use --no-reports for TravisCI testing
* Add possibility to download other genome for sareksnpeff and sarekvep containers
* Update docs
* Add --no-reports for all tests but MULTIPLE
* Use sensible label selectors for configs
* Enhance the old --noReports with --skip to skip QC on demand
* Replace --noReports by --skip all
* Update logos with new nf-core/sarek logo
* Add logo to MultiQC report
* Remove createMultiQCconfig()
* Improve docs about offline usage with bianca
* Code polishing
* Update CHANGELOG

Commit:
removed GeneSplicer

Commit:
changed Rscript path

Commit:
changed Rscript path

Commit:
changed StrelkaBP and VEP CPUs

maxulysse committed:
More fixes (#20)

* fix: fix logo
* fix: Fix path to reference genomes
* feat: add markdownlint config file
* fix: lint all md files + small typos corrected
* feat: update CHANGELOG

maxulysse committed:
Add Tiddit (#10)

* add tiddit to container

* add process for TIDDIT

* update TIDDIT to 2.7.1

* fix merge issues

* feat: add tests for TIDDIT

* feat: update CHANGELOG

* feat: Add documentation for TIDDIT

* fix: arrange TIDDIT output

* feat: filter out low quality calls from TIDDIT

* feat: complete TIDDIT docs

* fix: syntac error, bad grep

maxulysse committed:
More improvments (#21)

* feat: merge markdownlint config files

* fix: more linting on md files

* feat: add md linting to travisCI tests

* feat: sort tests

* feat: improve travis testing

* feat: improve Travis tests

* feat: sort tests

* feat: use YAML anchors to improve/simplify circle ci docker deployment

* fix: remove with stroke over aple core in nf-core logo

* feat: use direct link to slack sarek channel

* fix: code spacing

* feat: remove ALL from tests and simplify/improve script

* feat: use kraken profile to run tests

* feat: remove unnecessary Channel.create()

* fix: removed links to deleted files

* feat: moved smallGRCh37 to new genomes.config

* feat: update CHANGELOG

maxulysse committed:
Update docs (#29)

* feat: update docs
* feat: replace skip params by skipQC

alneberg committed:
Memory parameter singleCPUMem is now possible to use (#22)

* Memory parameter singleCPUMem is now possible to use
* Updated CHANGELOG
* No import, but in a new place

maxulysse committed:
Add GitHubActions (#27)

* feat: Add GitHubActions
* feat: reduce linting time
* feat: add --memory for script
* feat: use --max_memory to limit memory usage to 6.GB on GitHub actions
* install nextflow directly
* try matrix
* use export to get correct nextflow version
* add linting tests
* fix with node version indent
* rename jobs
* code polish
* feat: add badges for GitHub Actions
* test branch protection

maxulysse committed:
small refactoring (#30)

* Simplify code of MapReads process
* Fix launching of FastQCBAM process
* Add TIDDIT version to reports

maxulysse committed:
print more info for branch protection github actions

maxulysse committed:
adding nf-core lint (#31)

* Fix nf-core lint on sarek
* Remove extra CI from .travis.yml and GitHub Actions nf-core CI
* Add new nf-core extra CI
* remove fancy badges
* Remove TODO nf-core comments in code
* Use 1.1_2 version for rcolorbrewer, according to nf-core lint

maxulysse committed:
fix ci

maxulysse committed:
Install Ascat with bioconda (#32)

* install ASCAT in the main container
* update docs
* clean up environment.yml

maxulysse committed:
fix GitHub Actions CI

maxulysse committed:
Panel of Normals (#24)

* changed MuTect2 to Mutect2

* copied Mutect2 processes

* horrible mess right now

* fixed merge mixup

* looks Mutect2 works without PON

* added germline resource to igenomes

* we do know there are bugs around, but it is working

* MuTect2 -> Mutect2

* Looks PON works with both GRCh37 and 38

* added DOCS

* typo

* renamed GRCh37 intervals to wgs_calling_regions_Sarek.list

* PON in help and summary

* extended Mutect2 output

* even more extended Mutect2 output

* Commits for comments

* added contamination table to documents

* Update main.nf

Co-Authored-By: Maxime Garcia <max.u.garcia@gmail.com>

maxulysse committed:
fix MD lint

maxulysse committed:
got ascat version using R library(ASCAT) + use workflow.manifest.version to specify workflow version in path to R scripts for control-FREEC and VEP

maxulysse committed:
update CHANGELOG

maxulysse committed:
code polishing

maxulysse committed:
code polishing

maxulysse committed:
fix memory usage

apeltzer committed:
Merge pull request #33 from MaxUlysse/BackToLint

Fixing Lint

maxulysse committed:
update reference files

maxulysse committed:
fix params name

maxulysse committed:
fix branch protection GitHub actions

maxulysse committed:
fix branch protection GitHub actions

maxulysse committed:
fixing badges

maxulysse committed:
Merge remote-tracking branch 'upstream/dev' into AWS-iGenomes

maxulysse committed:
Change --sample by --input (#23)

* feat: remove unnecessary Channel.create()
* feat: replace --sample with --input, genomeFile with fasta...
* feat: update comments
* feat: update docs
* add deprecation message for --sample, --sampleDir, --anotateVCF, --noReports, --genomeDict, --genomeFile, --genomeIndex
    - Add message when deprecated params are used
    - Deprecated params are still working
    - Fix usage of deprecated params
* Fix --input for germline sample directory
* tsvPath is now defined as null by default
* add deprecation info in docs about --noReports

maxulysse committed:
Merge remote-tracking branch 'upstream/dev' into AWS-iGenomes

maxulysse committed:
Update tools

- `control-freec` from `11.4` to `11.5`
- `ensembl-vep` from `95.2` to `97.3`
- `freebayes` from `1.2.0` to `1.3.1`
- `gatk4` from `4.1.2.0` to `4.1.3.0`
- `igvtools` from `2.3.93` to `2.5.3`
- `manta` from `1.5.0` to `1.6.0`
- `qualimap` from `2.2.2b` to `2.2.2c`
- `vcfanno` from `0.3.1` to `0.3.2`

maxulysse committed:
fix path to files

apeltzer committed:
Merge pull request #34 from MaxUlysse/AWS-iGenomes

update reference files

maxulysse committed:
update CHANGELOG

maxulysse committed:
update files

maxulysse committed:
rename script

maxulysse committed:
Merge remote-tracking branch 'upstream/dev' into Updates

maxulysse committed:
code polishing

maxulysse committed:
revert updates

maxulysse committed:
revert updates

maxulysse committed:
update docs

maxulysse committed:
add comments to params

maxulysse committed:
remove Mutect2 from MULTIPLE

maxulysse committed:
more code polishing

maxulysse committed:
update CHANGELOG

Commit:
Merge remote-tracking branch 'nf-core/dev'

Conflicts:
	.github/CONTRIBUTING.md
	.github/ISSUE_TEMPLATE/bug_report.md
	.github/ISSUE_TEMPLATE/feature_request.md
	.github/PULL_REQUEST_TEMPLATE.md
	.gitignore
	.travis.yml
	CHANGELOG.md
	Dockerfile
	LICENSE
	README.md
	bin/concatenateVCFs.sh
	bin/convertAlleleCounts.r
	bin/run_ascat.r
	build.nf
	conf/base.config
	conf/genomes.config
	conf/igenomes.config
	docs/README.md
	docs/images/CAW_logo.png
	docs/images/CAW_logo.svg
	docs/images/NBIS_logo.png
	docs/images/NGI_logo.png
	docs/images/SciLifeLab_logo.png
	environment.yml
	main.nf
	nextflow.config

maxulysse committed:
more code polishing

maxulysse committed:
code polish

maxulysse committed:
revert changes

maxulysse committed:
update CHANGELOG

maxulysse committed:
Update conf/test.config

apeltzer committed:
Merge pull request #35 from MaxUlysse/Updates

Updates

Commit:
Merge remote-tracking branch 'nf-core/dev'

maxulysse committed:
move params definition from main.nf to conf/base.config

maxulysse committed:
default params.outdir is results

maxulysse committed:
update docs

maxulysse committed:
move all params from conf/base.config to nextflow.config

maxulysse committed:
enhanced config

maxulysse committed:
better logs

maxulysse committed:
code polishing

maxulysse committed:
add comments

apeltzer committed:
Initial template commit

apeltzer committed:
Merged vanilla TEMPLATE branch into main pipeline

maxulysse committed:
remove comments

maxulysse committed:
no need to use check_resource for cpus_max or memory_max

apeltzer committed:
Merge pull request #36 from MaxUlysse/CleanUp

Clean up

maxulysse committed:
Merge branch 'master' into dev

maxulysse committed:
fix markdownlint

maxulysse committed:
default PublishDirMode is now copy

maxulysse committed:
Prepare release (#37)

* nf-core bump-version . 2.5
* add --skipQC all to ANNOTATE test
* update usage of download_image script
* update CHANGELOG

maxulysse committed:
Update README.md

Co-Authored-By: Alexander Peltzer <apeltzer@users.noreply.github.com>

maxulysse committed:
Update README.md

maxulysse committed:
Apply suggestions from code review

Co-Authored-By: Alexander Peltzer <apeltzer@users.noreply.github.com>

maxulysse committed:
Apply suggestions from code review

maxulysse committed:
Avoid collision in MultiQC (#38)

* enforcing file names to be idsample_idRun pre merging to avoid collision
* more consistency for fastqc and bamqc output
* update CHANGELOG

maxulysse committed:
Fix ch_dbsnp channel (#39)

* fix typo for bwaIndex in params.genomes[params.genome]
* add more logs for reference Files
* fix params.dbsnpIndex
* remove bwaIndex from smallGRCh37
* fix ch_dbsnp
* update CHANGELOG

Commit:
Merge remote-tracking branch 'nf-core/dev'

maxulysse committed:
Update nextflow.config

maxulysse committed:
use tagged image

maxulysse committed:
add patch to branch protection

maxulysse committed:
remove VEP from CI as it's currently failing

maxulysse committed:
set time for MapReads process to 48.h

maxulysse committed:
Merge pull request #19 from nf-core/dev

First Release

maxulysse committed:
Add Zenodo Badge

maxulysse committed:
nf-core bump-version . 2.5.1dev

maxulysse committed:
Remove PublishDirMode from test profile (#40)

* remove PublishDirMode from test profile

maxulysse committed:
update all tools

maxulysse committed:
minor updates + typo fix (#42)

* minor updates + typo fix

maxulysse committed:
Merge pull request #41 from MaxUlysse/environmentUpdate

maxulysse committed:
fix VEP automated builds

maxulysse committed:
Merge pull request #43 from MaxUlysse/VEP

Fix automated VEP builds

maxulysse committed:
Worfklow (#45)

* Add workflow figure
* Include workflow figure in readme
* Update CHANGELOG

Commit:
Merge remote-tracking branch 'nf-core/dev'

Commit:
first version of transferring pathfindr scores to scout VCF

Commit:
Docs skeleton for downstream scores

Commit:
removed debug code, fixed columns

Commit:
extended docs

Commit:
additional documentation and fixed #CROM line

Commit:
better docs

MaxUlysse commented:
Closing because already ported [nf-core/sarek](https://github.com/nf-core/sarek/)
